numpy/lib/arraypad.py

```diff
@@ -294,27 +286,23 @@ def _append_ramp(arr, pad_amt, end, axis=-1):
     if pad_amt == 0:
         return arr
 
-    # Generate shape for final concatenated array
-    padshape = tuple(x if i != axis else pad_amt
-                     for (i, x) in enumerate(arr.shape))
-
-    # Generate an n-dimensional array incrementing along `axis`
-    ramp_arr = _arange_ndarray(arr, padshape, axis,
-                               reverse=False).astype(np.float64)
-
-    # Slice a chunk from the edge to calculate stats on
+    # Slice a chunk from the edge to calculate stats on and extract edge
     edge_slice = _slice_last(arr.shape, 1, axis=axis)
+    edge = arr[edge_slice]
+
+    ramp_arr = np.linspace(
+        start=end,
+        stop=edge.squeeze(axis),
+        num=pad_amt,
+        endpoint=False,
+        dtype=arr.dtype,
+        axis=axis
+    )
+    # Reverse linear space in appropriate dimension
+    ramp_arr = ramp_arr[
+        _slice_at_axis(ramp_arr.shape, slice(None, None, -1), axis)
+    ]
```

Comment:
We could avoid this if linspace had something like `startpoint=False` (similar to `endpoint=False`) the we could simply switch the start and stop values. IMO not worth it but I'd be interested to know if such a feature was discussed already.